client/components/data/query-canonical-theme/index.jsx

```diff
@@ -1,21 +1,32 @@
 import PropTypes from 'prop-types';
 import { Fragment } from 'react';
 import { connect } from 'react-redux';
-import QueryTheme from 'calypso/components/data/query-theme';
+import QuerySingleTheme from 'calypso/components/data/query-theme/query-single-theme';
 import { isWpcomTheme, isWporgTheme } from 'calypso/state/themes/selectors';
 import { knownConflictingThemes } from 'calypso/state/themes/selectors/get-canonical-theme';
 
 const QueryCanonicalTheme = ( { siteId, themeId, isWpcom, isWporg } ) => {
 	// Conflicting themes are themes we always search Jetpack+Atomic sites for information about.
 	// Usually, it's only searched if we can't find info on both Wpcom and Wporg.
 	const isConflictingTheme = knownConflictingThemes.has( themeId );
+
+	let sourceType = 'wpcom';
+
+	if ( isWporg && ! isWpcom ) {
+		sourceType = 'wporg';
+	}
+
+	if ( ! isWporg && ! isWpcom ) {
+		sourceType = 'jetpack';
+	}
+
+	if ( isConflictingTheme && siteId ) {
+		sourceType = 'jetpack';
+	}
+
```

Comment:
[Nitpick] This could be extracted to a function so that we can return instead of continuing to evaluate conditions and overwriting a variable.